core/src/main/scala/tech/beshu/ror/accesscontrol/audit/DataStreamAuditSinkCreator.scala

```diff
@@ -0,0 +1,175 @@
+  }
+
+  private def createIfNotExists(service: DataStreamService, settings: DataStreamSettings): Task[Unit] = {
+    service
+      .checkDataStreamExists(settings.dataStreamName)
+      .flatMap {
+        case true =>
+          Task.delay(logger.info(s"Data stream ${settings.dataStreamName.show} already exists"))
+        case false =>
+          setupDataStream(service, settings)
+      }
+  }
+
+  private def setupDataStream(service: DataStreamService, settings: DataStreamSettings): Task[Unit] = {
+    for {
+      _ <- Task.delay(logger.info(s"Trying to setup ROR audit sink with data stream ${settings.dataStreamName.show}.."))
+      _ <- service.createIndexLifecyclePolicy(settings.lifecyclePolicy, settings.lifecyclePolicyJson(service.capabilities.ilmMaxPrimaryShardSize))
+      _ <- service.createComponentTemplateForMappings(settings.componentTemplateMappingsId, settings.mappings, settings.mappingsMetadata)
+      _ <- service.createComponentTemplateForIndex(settings.componentTemplateSettingsId, settings.lifecyclePolicy, settings.indexSettingsMetadata)
+      _ <- service.createIndexTemplate(settings.indexTemplate, settings.dataStreamName, NonEmptyList.of(settings.componentTemplateMappingsId, settings.componentTemplateSettingsId), settings.indexTemplateMetadata)
+      _ <- service.createDataStream(settings.dataStreamName)
+      _ <- Task.delay(logger.info(s"ROR audit data stream ${settings.dataStreamName.show} created."))
+    } yield ()
+  }
+}
+
+
+object DataStreamAuditSinkCreator {
+
+  final case class DataStreamSettings(dataStreamName: DataStreamName.Full,
+                                      documentMappings: Map[String, FieldType]) {
+    def lifecyclePolicy: String = s"${dataStreamName.value.value}-lifecycle-policy"
+
+    def componentTemplateSettingsId: TemplateName = templateNameFrom(s"${dataStreamName.value.value}-settings")
+
+    def componentTemplateMappingsId: TemplateName = templateNameFrom(s"${dataStreamName.value.value}-mappings")
+
+    def indexTemplate: TemplateName = templateNameFrom(s"${dataStreamName.value.value}-template")
+
+    private def templateNameFrom(value: String) = {
+      TemplateName
+        .fromString(value)
+        .getOrElse(throw new IllegalStateException("Template name should be non-empty"))
+    }
+
+    val mappings: Value = serializeMappings(documentMappings)
+
+    val mappingsMetadata: Map[String, String] = {
+      Map(
+        "description" -> "Data mappings for ReadonlyREST audit data stream"
+      )
+    }
+
+    val indexSettingsMetadata: Map[String, String] = {
+      Map(
+        "description" -> "Index settings for ReadonlyREST audit data stream"
+      )
+    }
+
+    val indexTemplateMetadata: Map[String, String] = {
+      Map(
+        "description" -> "Index template for ReadonlyREST audit data stream"
+      )
+    }
+
+    def lifecyclePolicyJson(maxPrimaryShardSizeInRollover: Boolean): Value = {
```

Comment:
I'm not sure if we should do it like this. We define it at a core level, but it may change in the future. Maintaining compatibility at the core level will be difficult. At the same time, we have modules, where the compatibility is easier to maintain.